ersilia/publish/inspect.py

```diff
@@ -0,0 +1,359 @@
+from .. import ErsiliaBase
+import requests
+import subprocess
+import time
+import json
+from urllib.request import urlopen
+from ..hub.content.card import RepoMetadataFile
+
+
+class ModelInspector(ErsiliaBase):
+    def __init__(self, model, config_json=None):
+        ErsiliaBase.__init__(self, config_json=config_json, credentials_json=None)
+        self.model = model
+
+    def check_repo_exists(self, flag):
+        """
+        Verify that the repository exists at a given link.
+
+        Args:
+            flag (int): Flag indicating whether to return boolean or detailed message.
+
+        Returns:
+            bool or str: Depending on the flag, returns boolean or a message indicating the check status.
+        """
+        url = f"https://github.com/ersilia-os/{self.model}"
+        response = requests.head(url)
+        if flag == 0:
+            return response.status_code == 200
+        elif response.status_code == 200:
+            return "Check passed."
+        return f"Connection invalid, no github repository found at https://github.com/ersilia-os/{self.model}. Please check that this repository exists in the ersilia-os database."
+
+    def metadata_complete(self, flag):
```

Comment:
We can rename this to `check_complete_metadata`

Reply:
done